strapi/strapi
# fix:issue on do not validate locale if not creating for all locales

### What does it do?
After applying a change this [pull request ](https://github.com/strapi/strapi/pull/19626) some cases has been missed like for example when the local is not default one and you want to create and publish 

### Why is it needed?
Because the user can not create or save any single type  module so i just add another validation to check if the local exist or not

### How to test it?
Create single types and collection type with localization enabled , look to you database you can see all data will be create depend of your current local and when you are going to publish it it says 404 because it can not find the record with that local in the database

### Related issue(s)/PR(s)

- [#19798](https://github.com/strapi/strapi/issues/19798)
- [#19715](https://github.com/strapi/strapi/issues/19715)
-  [#19626](https://github.com/strapi/strapi/pull/19626) 
- [#19805](https://github.com/strapi/strapi/issues/19805)
